Make ChartViewBase's _data optional. (Fixes #771)

ospr opened the pull request:
Changed ChartViewBase's _data property declaration from _data! to _data? to allow for setting the chart's data back to nil. Updated the code base to property handle _data as an optional value.

danielgindi commented:
Looks nice. With some code moving from alpha Swift to modern Swift also...

How thoroughly was this tested?

ospr commented:
When I initially made this pull request I had verified that setting the BarLineChartViewBase object's data property to nil would cause it to behave properly and had verified that there were no other regressions by running through all of the charts in the demo project (along with toggling various options) and verifying that they behaved as expected.

However, I realized I forgot to test setting other chart type's data property to nil. I went through the demo project and added a "Toggle Data" option and then checked each chart type to verify that they behaved properly. I found a few minor bugs, fixed them, and squashed them back into this pull request. I made a separate request at #781 which includes the "Toggle Data" option if you wish to pull that in as well.

danielgindi commented:
Great! Thanks!
